descwl_coadd_task/tests/test_warps.py

```diff
@@ -0,0 +1,362 @@
+from __future__ import annotations
+
+import unittest
+
+import lsst.afw.cameraGeom.testUtils
+import lsst.afw.image as afw_image
+import lsst.skymap as skyMap
+import lsst.utils.tests
+import numpy as np
+from lsst.afw.detection import GaussianPsf
+from lsst.daf.butler import DataCoordinate, DimensionUniverse
+from lsst.pipe.base import InMemoryDatasetHandle
+from lsst.pipe.tasks.coaddBase import makeSkyInfo
+
+from descwl_coadd_task import MakeShearWarpConfig, MakeShearWarpTask
+
+PIXEL_SCALE = 0.2
+"""Pixel scale in arcseconds"""
+
+
+class MakeWarpTestCase(lsst.utils.tests.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        # Docstring inherited.
+        # This method contains the setup that needs to be done only once for
+        # all the tests.
+        cls.ny = 4000
+        cls.nx = 4072
+
+        cls._build_skyMap()
+        cls._generate_photoCalib()
+        cls.skyInfo = makeSkyInfo(cls.skyMap, 0, 36)
+
+        # Setup the metadata and empty exposures for warping.
+        cls._make_data()
+
+    def setUp(self):
+        # Docstring inherited.
+        # This method contains the setup that needs to be repeated for each
+        # test.
+        self.rng = np.random.default_rng(12345)
+
+        # Fill the input exposures with random data.
+        # This needs to be redone for each test, since the test may have
+        # modified the pixel values in-place.
+        for exposureRef in self.dataRefs:
+            self._fill_exposure(exposureRef.get())
```

Comment:
We can speed the tests further by filling the pixel values once in `setUpClass` and setting mask bits in the tests. We will still have to clear the masks in the `tearDown` method which gets run after every test.

Reply:
Its simpler and less error prone to remake the data each time rather than deal with what is effectively a global state paradigm.  I can look into it in a followup PR

Reply:
Just to be clear, `setUp` gets run before every test case. It's not global.

Reply:
I see.  I don't use this way of doing unit tests, I  just write functions `test_*` that get automatically run by `pytest`

How can you do different versions of setUp for different tests, do you need a new class?

What's the advantage of `setUp` versus having a function that creates the data, in which case you can send keywords to change the data.